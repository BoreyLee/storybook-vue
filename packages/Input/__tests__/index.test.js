import Element from "../src/index.vue";
import { mount } from "@vue/test-utils";

describe("Lbr-Input", () => {
  test("input-text", () => {
    const wrapper = mount(Element);
    expect(wrapper.html()).toContain('input type="text"');
  });
  test("input-password", () => {
    const wrapper = mount(Element, {
      propsData: { type: "password", value: "admin" },
    });
    expect(wrapper.html()).toContain('input type="password"');
    expect(wrapper.props("value")).toBe("admin");
  });
  /** 快照 */
  test("input-snapshot", () => {
    const wrapper = mount(Element, {
      propsData: { type: "password", value: "admin" },
    });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
