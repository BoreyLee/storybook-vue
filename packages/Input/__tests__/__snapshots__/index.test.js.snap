// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lbr-Input input-snapshot 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;

exports[`lbr-Input input-snapshot 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;
